tests/movies.spec.js

```diff
@@ -1,45 +1,36 @@
 import { expect, test } from '@jest/globals';
 import request from 'supertest';
 import app from '../app.js';
-
-test('Encanto page shows title of movie', async () => {
-  const response = await request(app).get('/movie/1').expect(200);
-
-  expect(response.text).toMatch('Isle of dogs');
-});
-
-test('Isle of dogs page shows title of movie', async () => {
-  const response = await request(app).get('/movie/2').expect(200);
-
-  expect(response.text).toMatch('Encanto');
-});
-
-test('The Shawshank Redemption page shows title of movie', async () => {
-  const response = await request(app).get('/movie/3').expect(200);
-
-  expect(response.text).toMatch('The Shawshank Redemption');
+import fetch from 'node-fetch';
+
+const url = 'https://plankton-app-xhkom.ondigitalocean.app/api/movies';
+const settings = { method: 'Get' };
+
+let APImapped;
+
+beforeAll(async () => {
+  const response = await fetch(url, settings);
+  const json = await response.json();
+
+  let APIdata = json;
+  APImapped = APIdata.data.map((movies) => {
+    return {
+      id: movies.id,
+      title: movies.attributes.title,
+      intro: movies.attributes.intro,
+      image: movies.attributes.image.url,
+    };
+  });
 });
 
-test('Min granne Totoro page shows title of movie', async () => {
-  const response = await request(app).get('/movie/4').expect(200);
+test(`page shows title of movie`, async () => {
+  for (let i = 0; i < APImapped.length; i++) {
```

Comment:
We generally avoid having logic (such as loops and conditionals) in our tests. We want our tests to verify that logic in the application code is correct, and if the tests contain logic (that could be incorrect) they become less reliable. At that point, you might want tests for your tests, and that's how you know you've gone wrong. 😊